src/examples/serial/serial.go

```diff
@@ -0,0 +1,10 @@
+package main
+
+import "time"
+
+func main() {
+	for {
+		println("hello world!")
+		time.Sleep(time.Second)
```

Comment:
I think this example was included by accident? It doesn't seem to belong in this PR (but a separate PR with this would be nice!).

Reply:
Yes that was an error. I can remove the file if you like and submit a separate PR, or just leave it. UP to you.

Reply:
Yes, if you want please remove it. Otherwise I can remove it when merging (or if you don't mind, send it in as a separate commit).

Offtopic: maybe you've tried to print multiple strings in a source file and found a crash. I'm aware of that, but it really is a bug in the AVR backend of LLVM. However, I plan to do an optimization at some point that should have the side effect of making this happen less often. Printing integers shouldn't result in such a crash.

Reply:
I noticed a different problem, that this did not work:

```go
val := uint16(1000)
println("value is", val)
```

The output was:
```
255
```

If I did same but using uint8 it worked as expected.

Reply:
That's a bug, I'll have to look into it.

Reply:
OK removed the serial example. I will submit a separate PR.

Reply:
My example above to repro the `println()` bug was flawed. Now that I am back at computer, here is a way to repro using this branch:

```go
package main

import (
	"machine"
	"time"
)

func main() {
	machine.InitADC()

	led := machine.GPIO{machine.LED}
	led.Configure(machine.GPIOConfig{Mode: machine.GPIO_OUTPUT})

	sensor := machine.ADC{machine.ADC2}
	sensor.Configure()

	for {
		val := sensor.Get()
		if val < 512 {
			led.Low()
		} else {
			led.High()
		}

		println("Value is", val) // this prints only the numeric value of val, not the string prefix "Value is"
		time.Sleep(time.Millisecond * 100)
	}
}
```